waku/waku_lightpush/protocol.nim

```diff
@@ -54,24 +54,31 @@ proc initProtocolHandler*(wl: WakuLightPush) =
     let
       pubSubTopic = req.request.get().pubSubTopic
       message = req.request.get().message
-    debug "push request", peerId=conn.peerId, requestId=req.requestId, pubsubTopic=pubsubTopic
-
+      contentTopic = message.contentTopic
+    debug "push request", peerId=conn.peerId, requestId=req.requestId, pubsubTopic=pubsubTopic, contentTopic=contentTopic
+    
     var response: PushResponse
-    var handleRes: WakuLightPushResult[void]
-    try:
-      handleRes = await wl.pushHandler(conn.peerId, pubsubTopic, message)
-    except Exception:
-      response = PushResponse(is_success: false, info: some(getCurrentExceptionMsg()))
-      waku_lightpush_errors.inc(labelValues = [messagePushFailure])
-      error "pushed message handling failed", error= getCurrentExceptionMsg()
-
-
-    if handleRes.isOk():
-      response = PushResponse(is_success: true, info: some("OK"))
+    
+    if contentTopic == "":
+      response = PushResponse(is_success: false, info: some(emptyContentTopicFailure))
+      waku_lightpush_errors.inc(labelValues = [emptyContentTopicFailure])
+      error "empty content topic", error=emptyContentTopicFailure
```

Comment:
I don't quite interpret the bug in https://github.com/waku-org/nwaku/issues/1641 in this way. The issue is not that the content topic is an "empty string" it's that that field is _unset_ in the protocol buffer, resulting in a wrongly encoded RPC that fails to decode. The same issue states:

> Send a light push rpc request that contains a missing protobuf field such as content topic

So it's not related to the content topic specifically, but any missing field that causes an rpc decode failure. Since we still return in lines 45 and 51 on invalid RPC (protobufs), afaict the original issue won't be fixed.

Reply:
Thank you, now I think i understand the issue properly!

If we want to return a `PushResponse` after a decoding failure, what should its `requestId` be set to? Normally, it is copied from the corresponding field in `PushRequest`, but if the request cannot be decoded properly, we can't count on that. Can we just omit the `requestId` field in that case? It is not wrapped in `Option` in the definition of `PushRPC`... Or shall we use some default value?

Reply:
Mmm. This is a good point. I would actually guess that in almost all cases decoding fail because of an inner field (either `PushRequest` or `WakuMessage`) failing to decode, rather than the outer `PushResponse`. In other words, it may be possible to change to a two stage decoding process where the `requestId` can be extracted before attempting to decode the encapsulated request.

However, for now this may be overkill and the best option is probably to initialise the `requestId` as an empty string.

@fryorcraken @danisharora099 since the PushRPC and PushRequest fail to decode properly in this bug, it's difficult to extract a `requestId`. Would a decent solution (from js-waku's perspective) be to return a well-formed response, with error code, but with the `requestId` set to an empty string?

Reply:
> @fryorcraken @danisharora099 since the PushRPC and PushRequest fail to decode properly in this bug, it's difficult to extract a `requestId`. Would a decent solution (from js-waku's perspective) be to return a well-formed response, with error code, but with the `requestId` set to an empty string?

This is a fair solution IMO.

We currently do not use the `requestId`, we are able to assume the response is on the same multiplex stream than the request (or at least it translates this way in js-libp2p API).